### PR TITLE
acts dependencies: new versions as of 2024/12/08

### PR DESCRIPTION
**Pull request by stephenswat**
This commit includes a new version of ACTS, as well as new versions of the ACTS algebra plugins, covfie, detray, and GeoModel.

### CODE REVIEW

File: var/spack/repos/builtin/packages/acts/package.py [hunk 1]
```diff
@@ -41,6 +41,7 @@ class Acts(CMakePackage, CudaPackage):
     # Supported Acts versions
     version("main", branch="main")
     version("master", branch="main", deprecated=True)  # For compatibility
+    version("38.1.0", commit="8a20c88808f10bf4fcdfd7c6e077f23614c3ab90", submodules=True)
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/acts/package.py:44]: https://github.com/acts-project/acts/pull/3921 implies that acts@:38.0 is not supporting edm4hep@0.99: so we may need to add a conflict (probably easier than adding this through version ranges).

do you have any intentation to keep track of compiler compatibility, e.g. https://github.com/acts-project/acts/pull/3938?

File: var/spack/repos/builtin/packages/acts-algebra-plugins/package.py [hunk 2]
```diff
@@ -17,6 +17,7 @@ class ActsAlgebraPlugins(CMakePackage):
 
     license("MPL-2.0", checked_by="stephenswat")
 
+    version("0.26.0", sha256="301702e3d0a3d12e46ae6d949f3027ddebd0b1167cbb3004d9a4a5697d3adc7f")
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/acts-algebra-plugins/package.py:20]: 0.26.0 seems to require C++20, not just "allow" it, see https://github.com/acts-project/algebra-plugins/pull/128/files#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20a

This package recipe will still set this to cxxstd=17 by default.

File: var/spack/repos/builtin/packages/covfie/package.py [hunk 3]
```diff
@@ -20,6 +20,7 @@ class Covfie(CMakePackage, CudaPackage):
 
     maintainers("stephenswat")
 
+    version("0.11.0", sha256="39fcd0f218d3b4f3aacc6af497a8cda8767511efae7a72b47781f10fd4340f4f")
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/covfie/package.py:23]: CMake minimum is now 3.21.

COVFIE_REQUIRE_CXX20 is removed and c++20 is now required, https://github.com/acts-project/covfie/compare/v0.10.0...v0.11.0#diff-5e713e1875f76db7b6b0a94d498a608b1b0ce4295abefc00287eb649e912e367. I think the `concepts` variant should be removed, or set to default=True now (removing a variant messes with configs that have `covfie +concepts` which will effectively remain pinned at 0.10.0).

**Review by wdconinc — COMMENTED** [review]
Verified checksums and hashes. A few comments.

**Comment by stephenswat** [comment]
Pinging @paulgessinger as my fellow ACTS Spack enthusiast.

File: var/spack/repos/builtin/packages/acts/package.py (hunk 1 above)
Comment (stephenswat) [reply var/spack/repos/builtin/packages/acts/package.py:44]: Updated by adding a conflict as you suggested.

File: var/spack/repos/builtin/packages/acts-algebra-plugins/package.py (hunk 2 above)
Comment (stephenswat) [reply var/spack/repos/builtin/packages/acts-algebra-plugins/package.py:20]: Removed the C++17 default for 0.26.0: using the same mechanism as in ACTS.

File: var/spack/repos/builtin/packages/covfie/package.py (hunk 3 above)
Comment (stephenswat) [reply var/spack/repos/builtin/packages/covfie/package.py:23]: I just removed the variant all together. Given the very small userbase of this package I thought that would be fine.

File: var/spack/repos/builtin/packages/geomodel/package.py [hunk 4]
```diff
@@ -18,6 +18,7 @@ class Geomodel(CMakePackage):
 
     license("Apache-2.0", checked_by="wdconinc")
 
+    version("6.7.0", sha256="bfa69062ba191d0844d7099b28c0d6c3c0f87e726dacfaa21dba7a6f593d34bf")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/geomodel/package.py:21]: I cannot confirm this due to `read operation timed out`.

**Review by tldahlgren — COMMENTED** [review]
Confirmed the version sha256s/commit *except* that of `geomodel`.

File: var/spack/repos/builtin/packages/covfie/package.py (hunk 3 above)
Comment (wdconinc) [reply var/spack/repos/builtin/packages/covfie/package.py:23]: Removing variant altogether is fine. Did you want to update the cmake minimum?

File: var/spack/repos/builtin/packages/covfie/package.py [hunk 5]
```diff
@@ -20,19 +20,17 @@ class Covfie(CMakePackage, CudaPackage):
...
     version("0.10.0", sha256="d44142b302ffc193ad2229f1d2cc6d8d720dd9da8c37989ada4f23018f86c964")
 
     depends_on("cxx", type="build")  # generated
 
-    variant("concepts", default=False, description="Enforce C++20 concepts")
-
     depends_on("cmake@3.18:", type="build")
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/covfie/package.py:29]:
```suggestion
    depends_on("cmake@3.21:", type="build", when="@0.11:")
    depends_on("cmake@3.18:", type="build")
```

File: var/spack/repos/builtin/packages/geomodel/package.py (hunk 4 above)
Comment (wdconinc) [reply var/spack/repos/builtin/packages/geomodel/package.py:21]: Might have been a transient.

```
$ spack checksum --verify --batch geomodel 6.7.0
==> Found 1 version of geomodel
==> Fetching https://gitlab.cern.ch/GeoModelDev/GeoModel/-/archive/6.7.0/GeoModel-6.7.0.tar.bz2
==> Verified 1 of 1
  
  6.7.0  [=] Correct
  
```

**Review by wdconinc — APPROVED** [review]
Thans for the updates.

I'll wait until @tldahlgren confirms there are no issues pulling the tarball, since sites like mine might already have incomplete certificate chains accessible in local cache and therefore not show some issues.

File: var/spack/repos/builtin/packages/geomodel/package.py (hunk 4 above)
Comment (tldahlgren) [reply var/spack/repos/builtin/packages/geomodel/package.py:21]: 🤔 I still get read operation timed out. Perhaps it's related to being on VPN?

**Review by tldahlgren — APPROVED** [review]
I can't confirm `geomodel` change but, since @wdconinc can, I'll approve and enable merging.

**Pull request merged**